### PR TITLE
[auto-ts] add memory check

### PR DESCRIPTION
**Pull request by stepanblyschak**
#### Why I did it

To support automatic techsupport invokation in case memory usage is too high.

#### How I did it

Implemented according to https://github.com/Azure/SONiC/pull/939

#### How to verify it

UT, manual test on the switch.

*DEPENDS* on https://github.com/Azure/sonic-utilities/pull/2116

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

### CODE REVIEW

**Review by vivekreddynv — DISMISSED** [review]

**Comment by liat-grozovik** [comment]
@qiluo-msft could you please help to review or suggest someone who should? This is a feature for 202205

**Comment by stepanblyschak** [comment]
@liat-grozovik It makes sense to run the checks after the dependency PR is merged and utilities submodule is updated

File: src/sonic-yang-models/yang-models/sonic-auto_techsupport.yang [hunk 1]
```diff
@@ -59,6 +59,18 @@ module sonic-auto_techsupport {
...
+                        description "Memory threshold; 0 to disable techsupport invocation on memory usage threshold crossing";
+                        type decimal-repr;
+                        default 10.0;
+                    }
+                    
+                    leaf min_available_mem {
+                        description "Minimum Free memory (in MB) that should be available for the techsupport execution to start";
+                        type uint32;
+                        default 200;
+                    }
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-auto_techsupport.yang:73]: Can you add unit test for these two new fields?

**Review by ganglyu — APPROVED** [review]

**Comment by yxieca** [comment]
@stepanblyschak I've reverted this change in 202205 branch because it is causing build failure. Is this a feature or bug fix? Why do we need it in 202205 branch?

**Pull request merged**